lib/python/qmk/cli/new/keyboard.py

```diff
@@ -0,0 +1,182 @@
+"""This script automates creating the files necessary for a new QMK keyboard.
+"""
+import shutil
+import hashlib
+import re
+
+from milc import cli
+from pathlib import Path
+from datetime import datetime
+
+
+@cli.argument('-kb', '--keyboard', help='Specify keyboard name. Example: clueboard/66')
+@cli.argument('-pr', '--project', help='The human-friendly name of the keyboard. Example: Clueboard 66%')
+@cli.argument('-mcu', '--microcontroller', help='Specify the microcontroller used for the keyboard.')
+@cli.argument('-u', '--username', help='Your GitHub username. Will be pasted into generated files.')
+@cli.subcommand('Creates a new keyboard project.')
+
+def new_keyboard(cli):
+    """Creates a new keyboard project.
+    """
+
+    # Root path for template files
+    template_root_path = Path("quantum/template")
+    template_base_path = Path(template_root_path) / "base"
+    year = datetime.now().year
+
+    # valid microcontroller choices
+    mcus = (
+        ( 'at90usb1286', 'avr' ),
+        ( 'at90usb646' , 'avr' ),
+        ( 'atmega16u2' , 'avr' ),
+        ( 'atmega16u4' , 'avr' ),
+        ( 'atmega328p' , 'avr' ),
+        ( 'atmega32a'  , 'ps2avrgb' ),
+        ( 'atmega32u2' , 'avr' ),
+        ( 'atmega32u4' , 'avr' ),
+        ( 'STM32F042'  , 'stm32' ),
+        ( 'STM32F072'  , 'stm32' ),
+        ( 'STM32F103'  , 'stm32' ),
+        ( 'STM32F303'  , 'stm32' ),
+    )
```

Comment:
This should live inside `qmk.constants`. For flexibility and use by other tools we may want to make it a different datastructure.